delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/FetchResource.scala

```diff
@@ -0,0 +1,29 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.resources
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Fetch.FetchF
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.model.{Resource, ResourceState}
+import ch.epfl.bluebrain.nexus.delta.sourcing.Transactors
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{ProjectRef, ResourceRef}
+import ch.epfl.bluebrain.nexus.delta.sourcing.state.ScopedStateGet
+import doobie.implicits._
+import doobie.{Get, Put}
+
+trait FetchResource {
+
+  def latest(ref: ResourceRef, project: ProjectRef): FetchF[Resource]
+
+}
+
+object FetchResource {
+
+  implicit private val getValue: Get[ResourceState] = ResourceState.serializer.getValue
+  implicit private val putId: Put[Iri]              = ResourceState.serializer.putId
+
+  def apply(xas: Transactors): FetchResource = (ref: ResourceRef, project: ProjectRef) =>
+    ScopedStateGet
+      .latest(Resources.entityType, project, ref.iri)
```

Comment:
Depending on the case you may want to resolve any type of resource

Reply:
Does this imply we should have a function in `ScopedStateGet` that allows fetching the state without specifying the entity type?

Reply:
So far, we have to focus on the one related to resources/schemas, the general one is only used for resolver endpoints afaik so we should just leave it as it is for now as we don't need it for the batch